addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2334,16 +2337,27 @@ export class OdooEditor extends EventTarget {
             // clean its children.
             for (const attribute of [...node.attributes]) {
                 // Keep allowed styles on nodes with allowed tags.
-                if (CLIPBOARD_WHITELISTS.styledTags.includes(node.nodeName) && attribute.name === 'style') {
-                    const spanInlineStyles = attribute.value.split(';').map(x => x.trim());
-                    const allowedSpanInlineStyles = spanInlineStyles.filter(rawStyle => {
-                        const [styleName, styleValue] = rawStyle.split(':');
-                        const style = CLIPBOARD_WHITELISTS.styles[styleName.trim()];
-                        return style && !style.defaultValues.includes(styleValue.trim());
-                    });
+                if (CLIPBOARD_WHITELISTS.styledTags.includes(node.nodeName) 
+                        && (attribute.name === 'style' 
+                            || attribute.name in CLIPBOARD_WHITELISTS.styles
+                            || attribute.name in CLIPBOARD_WHITELISTS.styleMap)) {
+                    let allowedSpanInlineStyles;
+                    if (attribute.name === 'style') {
+                        // If dealing with the (inline) style attribute, filter its values
+                        const spanInlineStyles = attribute.value.split(';').map(x => x.trim());
+                        allowedSpanInlineStyles = spanInlineStyles.filter(rawStyle => {
+                            const [styleName, styleValue] = rawStyle.split(':');
+                            const style = CLIPBOARD_WHITELISTS.styles[styleName.trim()];
+                            return style && !style.defaultValues.includes(styleValue.trim());
+                        }).join(';');
+                    } else {
+                        // If dealing with another (non-inline, style-ish) supported attribute, add it to the inline style values
+                        const styleName = CLIPBOARD_WHITELISTS.styleMap[attribute.name] || attribute.name
+                        allowedSpanInlineStyles = `${node.getAttribute('style') || ''} ${styleName}:${attribute.value};`
+                    }
                     node.removeAttribute(attribute.name);
                     if (allowedSpanInlineStyles.length > 0) {
-                        node.setAttribute(attribute.name, allowedSpanInlineStyles.join(';'));
+                        node.setAttribute('style', allowedSpanInlineStyles);
                     } else if (['SPAN', 'FONT'].includes(node.tagName)) {
                         for (const unwrappedNode of unwrapContents(node)) {
```

Comment:
I'm kind of worried that adding more supported styles will make some sub-nodes not be "unwrapped". That's why I was asking questions about this else statement. This works and produces seemingly correct output in my testing but I'm testing tables and probably don't foresee every case this method is used.

Reply:
If I understand correctly, the reason for `'SPAN', 'FONT'` is that other tags that can enter this block (such as `'B', 'I'`) don't have children node (other than text).
If we later add `TD` to `styledTags` it should also be added here IMO.